### PR TITLE
machines: Fix displaying capacity for Disks

### PR DESCRIPTION
**Pull request by mareklibra**
Bug: the Allocation/Capacity column were not rendered if some disk stats
are NaN (like in case of CDROM).

This patch provides fix.

### CODE REVIEW

**Comment by dperpeet** [comment]
I think it would be a good idea to add some checking of the stats to the integration test - check known capacity infos for example.

**Comment by mareklibra** [comment]
@dperpeet , it's already there, please search for "{0}-disks-{1}-capacity" or "{0}-disks-{1}-used" strings in check-machines.

This bug was not caught by the integration test because CDROM is not used for in the dummy test VM (just regular disks). i don't know how to fake it without provided an .iso . Moreover, there are plenty of disk types, we can't cover all of them.

The bug appeared in a former "fix" handling unsupported functionality for old Libvirt version.

File: pkg/machines/vmdiskstab.jsx [hunk 1]
```diff
@@ -120,22 +120,19 @@ const VmDisksTab = ({ vm, provider }) => {
         supported, but not available yet ~ will be read soon
         unsupported ~ failed attempt to read (old libvirt)
      */
-    let areDiskStatsSupported = true;
-    let showDiskStatsColumns = false;
+    let areDiskStatsSupported = false;
     if (vm.disksStats) {
-        const isNaNFound = Object.getOwnPropertyNames(vm.disksStats).some(target => { // look for NaN value
+        const isNonNaNFound = Object.getOwnPropertyNames(vm.disksStats).some(target => { // look for non-NaN value (means disk stats at least partially read)
```
Comment (dperpeet) [inline pkg/machines/vmdiskstab.jsx]: how about renaming this to something like `hasStatistics` and comment `// we have stats if there is at least one non-NaN value`
"non-NaN" is a double negative

File: pkg/machines/vmdiskstab.jsx [hunk 2]
```diff
@@ -120,22 +120,19 @@ const VmDisksTab = ({ vm, provider }) => {
...
             if (!vm.disksStats[target]) {
                 return false; // not yet retrieved, can't decide about disk stats support
             }
-            showDiskStatsColumns = true; // at least one disk stats read attempt finished
-            return isNaN(vm.disksStats[target].capacity); // if true, disk stats read attempt executed, but failed
+            return !isNaN(vm.disksStats[target].capacity) || !isNaN(vm.disksStats[target].allocation);
         });
-        areDiskStatsSupported = !isNaNFound;
-        showDiskStatsColumns = showDiskStatsColumns && areDiskStatsSupported;
+        areDiskStatsSupported = !!isNonNaNFound;
```
Comment (dperpeet) [inline pkg/machines/vmdiskstab.jsx]: why not assign this directly here if `isNonNaNFound` isn't used in other places anyway...

Comment (mareklibra) [inline pkg/machines/vmdiskstab.jsx]: Done

File: pkg/machines/vmdiskstab.jsx (hunk 1 above)
Comment (mareklibra) [inline pkg/machines/vmdiskstab.jsx]: Done

**Pull request merged**